mlflow/models/python_api.py

```diff
@@ -142,6 +143,13 @@ def predict(
                 the logged model's dependency using `mlflow.models.update_model_requirements` API
                 without re-logging it. Note that a registered model is immutable, so you need to
                 register a new model version with the updated model.
+        extra_envs: If specified, a dictionary of extra environment variables will be passed to the
+            model inference environment. This is useful for testing what environment variables are
+            needed for the model to run correctly. By default, environment variables existing in the
+            current os.environ are passed, and this parameter can be used to override them.
+
+            .. note::
```

Comment:
Can we raise if env_manager != _EnvManager.VIRTUAL_ENV and this argument is set? 
We should not permit users to get the impression that setting this param with an unsupported env management framework will work - they'll get super confused if they're supplying the ENV VARs when selecting a different backend and having the values not picked up properly and throwing an auth error. 
We should just explicitly forbid this and explain to the users that they need to use VIRTUAL_ENV with env var options